c10/util/Exception.cpp

```diff
@@ -64,16 +64,20 @@ void Error::AppendMessage(const std::string& new_msg) {
 }
 
 void Warning::warn(SourceLocation source_location, std::string msg) {
-  warning_handler_(source_location, msg.c_str());
+  warning_handler_(source_location, msg);
```

Comment:
nit: `std::move(msg)` here

Reply:
Actually, it looks like the warning handler takes this by const reference. In that case, warn() should also take by const reference...